elf/io/intern_wrapper.py

```diff
@@ -0,0 +1,76 @@
+from collections.abc import Mapping
```

Comment:
This file is just a copy of the same wrapper code for MRC.

Reply:
Ok, I am not really sure if / how much we need to copy here; depends a bit on how the intern api is structured.
Does it contain the `path/internal_path` distinction of hdf5/n5/zarr? MRC doesn't (because one mrc file only contains a single volume) so the internal path is hard-coded to `data` here.

Reply:
For write access you would also need to add  `__set_item__` here.

Reply:
Thinking about this more: we probably do need this wrapper, unless `intern` has a class that provides `dtype`, `ndim`, `chunks`, `shape`, `size` and `attrs` as properties.

Reply:
Awesome — yes, unfortunately the `intern` library doesn't have something that implements the `chunks`/`ndim`/`size` properties (though it should!). It does have something that implements `__getitem__` and `__setitem__` (and `shape`/`dtype`) with numpy syntax though:

https://github.com/jhuapl-boss/intern/wiki/Using-the-intern.array-API

Much like MRC, a single cloud protocol URI points to a single volume. So `bossdb://collectionname/experimentname/channelname` is a single 3D or 4D dataset; I think it's fair to mirror the MRC behavior for now.
(Technically, for BossDB this type of dataset/group organization DOES exist, in the form of collections/experiments/channels... but I think it makes sense to keep things general in this implementation, to work with other cloud datastores too. So ignore this parenthetical, ha!)